darkcluster/src/main/java/com/linkedin/darkcluster/impl/ConstantMultiplierDarkClusterStrategy.java

```diff
@@ -0,0 +1,118 @@
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.darkcluster.impl;
+
+import java.util.Random;
+
+import javax.annotation.Nonnull;
+
+import com.linkedin.common.util.Notifier;
+import com.linkedin.d2.balancer.ServiceUnavailableException;
+import com.linkedin.d2.balancer.util.ClusterInfoProvider;
+import com.linkedin.darkcluster.api.BaseDarkClusterDispatcher;
+import com.linkedin.darkcluster.api.DarkClusterStrategy;
+import com.linkedin.r2.message.RequestContext;
+import com.linkedin.r2.message.rest.RestRequest;
+
+/**
+ * ConstantMultiplierDarkClusterStrategy figures out how many dark requests to send. It uses the {@link ClusterInfoProvider} to determine the number of
+ * instances in both the source and target cluster, and uses that to calculate the number of request to send in order to make the level of traffic
+ * proportional to itself on any instance in the dark cluster (accounting for multiplier), assuming all hosts in the source cluster send traffic.
+ */
+public class ConstantMultiplierDarkClusterStrategy implements DarkClusterStrategy
+{
+  private final String _originalClusterName;
+  private final String _darkClusterName;
+  private final Float _multiplier;
+  private final BaseDarkClusterDispatcher _baseDarkClusterDispatcher;
+  private final Notifier _notifier;
+  private final Random _random;
+  private final ClusterInfoProvider _clusterInfoProvider;
+
+  public ConstantMultiplierDarkClusterStrategy(@Nonnull String originalClusterName, @Nonnull String darkClusterName, @Nonnull Float multiplier,
+                                               @Nonnull BaseDarkClusterDispatcher baseDarkClusterDispatcher,
+                                               @Nonnull Notifier notifier,
+                                               @Nonnull ClusterInfoProvider clusterInfoProvider,
+                                               @Nonnull Random random)
+  {
+    _originalClusterName = originalClusterName;
+    _darkClusterName = darkClusterName;
+    _multiplier = multiplier;
+    _baseDarkClusterDispatcher = baseDarkClusterDispatcher;
+    _notifier = notifier;
+    _clusterInfoProvider = clusterInfoProvider;
+    _random = random;
+  }
+
+  @Override
+  public boolean handleRequest(RestRequest originalRequest, RestRequest darkRequest, RequestContext requestContext)
+  {
+    int numRequestDuplicates = getNumDuplicateRequests(_random.nextFloat());
+    return _baseDarkClusterDispatcher.sendRequest(originalRequest, darkRequest, requestContext, numRequestDuplicates);
+  }
+
+  /**
+   * The formula to keep traffic proportional to the sending cluster is
+   * Avg#DarkRequests = ((# instances in dark cluster) * multiplier) / (# instances in source cluster)
```

Comment:
Yep - sorry forgot to remove this comment after we discussed. Let me think about naming. Not a huge fan of RelativeTraffic either since it's still not obvious that traffic is kept proportional for individual hosts.